### DISCUSSION TITLE
IP Protection

### DISCUSSION THREAD

**Discussion by rockbo842** [Tzahi12345/YoutubeDL-Material#432] [General] [answered]
So, I love this and thanks for making it!

The problem I ran into is that I was overzealous, added a bunch of channels that I like all at once, and promptly had my IP blocked.  Luckily, I was using a VPN.  Is there a way to set the behavior so that it is more conservative and doesn't throw up red flags?  For example, telling it how many videos it can pull in an hour, randomizing the times between downloads, etc.?

I'm running this on Unraid and I'm about as noob as you can get, so if there is a way to do this, you'd have to explain it to me like I'm 3=)

**Comment by Tzahi12345** [comment, marked as answer]
There's a couple settings that you'll only see in the `nightly` version, but they should help solve this problem:

![image](https://user-images.githubusercontent.com/707413/130546011-9cccbfd8-3019-42a9-ac9f-2c5f75429926.png)

Download delay adds a delay in seconds in between video downloads within a subscription. I've seen recommendations on this setting at around 60 or so.

The check interval is the overall amount of time it takes for the server to check all subscriptions, so increasing this will make YT less angry and less likely to ban your IP.

**Reply by rockbo842** [reply]
Thanks for the reply!  Again, I'm a noob so take that into account with the following:

1. If I'm using the docker container available in the Unraid Community Apps, am I able to use the nightly build?
2. The download delay is per subscription.  Does the app download from multiple subscriptions simultaneously?  Is there a config option to set the number of simultaneous downloads the app can do overall?

Thanks again!

**Reply by GlassedSilver** [reply]
1: Yes, you can use the nightly in it.
2. Yes that’s per subscription and yes it does check them all at once, however with more moderate settings I’m seeing much less errors. Mind you, adding a lot of subscriptions at once is kinda asking for trouble a little bit as of now. Ideally you add them spaced out as things stand now, maybe it would be less problematic though if you’re on the latest nightly. To get there edit the youtubedl-material template by adding a `:nightly` to the repository used.
![B60146B4-015E-402E-A9BF-63BAEC83A593](https://user-images.githubusercontent.com/1912133/130643868-4508d2cd-2ad9-4090-950b-c64aea8f80e7.jpeg)

**Reply by rockbo842** [reply]
> 
> 
> 1: Yes, you can use the nightly in it.
> 2. Yes that’s per subscription and yes it does check them all at once, however with more moderate settings I’m seeing much less errors. Mind you, adding a lot of subscriptions at once is kinda asking for trouble a little bit as of now. Ideally you add them spaced out as things stand now, maybe it would be less problematic though if you’re on the latest nightly. To get there edit the youtubedl-material template by adding a `:nightly` to the repository used.
> ![B60146B4-015E-402E-A9BF-63BAEC83A593](https://user-images.githubusercontent.com/1912133/130643868-4508d2cd-2ad9-4090-950b-c64aea8f80e7.jpeg)

Thanks!  I was able to get the nightly with your assistance.  We will see if I can keep my IP from getting blocked=)
